### PR TITLE
TCH-22 Add new tabs for challenges

### PR DESCRIPTION
**Pull request by adrian-musora**
**Design:**
- moved contentMetaData into a separate file
- refactored to have common elements in the `commonMetaData` object to avoid duplication
- added Challenges Tabs
-- these don't work as they require `fetchAll` to be refactor to handle group by on non-references.

**Testing**
- added MetaDataTests, didn't add anything for challenges.

### CODE REVIEW

File: src/contentMetaData.js [hunk 1]
```diff
@@ -0,0 +1,1154 @@
...
+            allowableFilters: ['genre', 'focus'],
+            sortBy: '-published_on',
+    },
+    'challenge': {
+        name: 'Challenges',
+            icon: 'icon-courses',
+            description: "... ",
+            allowableFilters: ['difficulty', 'topic', 'genre'],
+            sortBy: '-published_on',
+            modalText: 'Challenges are a collection of Workout-style videos that build your skills one step at a time. They help you develop broader musical skills at a manageable pace — usually over a few days.',
```
Comment (MeenuDhanoa) [inline src/contentMetaData.js:18]:
```suggestion
            modalText: 'Challenges are a series of guided lessons designed to build your skills day-by-day.',
```

**Review by MeenuDhanoa — APPROVED** [review]

File: src/contentMetaData.js [hunk 2]
```diff
@@ -0,0 +1,1154 @@
...
+                },
+                {
+                    name: 'Genres',
+                    short_name: 'Genres',
+                    is_group_by: true,
+                    value: 'genre',
+                },
+            ],
+        },
+        'challenges': {
```
Comment (MeenuDhanoa) [inline src/contentMetaData.js:298]: I believe this isn’t needed anymore since we’ve added the 'challenge' data type. Both would serve the same purpose.

Comment (adrian-musora) [reply src/contentMetaData.js:298]: These are a distinct content type specifically for drumeo. Roxana would be a better person to ask regarding history. For the moment these will stay in the system, though I may remove them when we do the data import/update later depending on how things fit together.

File: src/contentMetaData.js [hunk 3]
```diff
@@ -0,0 +1,1154 @@
...
+                    value: 'genre',
+                },
+            ],
+        },
+    },
+    'guitareo': {
+        'instructor': {
+            description: "Tackle your next guitar goal with bite-sized courses from many of the world's best guitarists.",
+        },
+        'courses': {
```
Comment (roxanariza) [inline src/contentMetaData.js:882]: Invalid key. Should remain ''course' like on drumeo/pianote

File: src/contentMetaData.js [hunk 4]
```diff
@@ -0,0 +1,1154 @@
...
+        'student-review': {
+            thumbnailUrl: 'https://d1923uyy6spedc.cloudfront.net/student-reviews-singeo.png',
+            description: "Want feedback on your playing? Submit a video for student review. We will watch your submission and then provide helpful encouragement and feedback. This is a great way to build accountability and benefit from the expertise of our teachers.",
+        },
+    },
+    'singeo':{
+        'instructor': {
+            description: "Your singing journey is unique. You need personalized coaching that helps you reach your goals. Learn from some of the best singers and vocal coaches in the world!",
+        },
+        'courses': {
```
Comment (roxanariza) [inline src/contentMetaData.js:990]: Invalid key. Should remain ''course' like on drumeo/pianote

File: src/contentMetaData.js [hunk 5]
```diff
@@ -0,0 +1,1154 @@
...
+        'instructor': {
+            description: "Your drumming journey is unique. You need personalized coaching that helps you reach your goals. Learn from some of the best drummers in the world!",
+        },
+        'course': {
+            name: 'Courses',
+            icon: 'icon-courses',
+            description: "Tackle your next drumming goal with bite-sized courses from many of the world's best drummers.",
+            allowableFilters: ['difficulty', 'genre', 'essential', 'theory', 'lifestyle', 'creativity'],
+            sortBy: '-published_on',
+            tabs: [
```
Comment (roxanariza) [inline src/contentMetaData.js:183]: In general the tabs are the same for content types per each brand; there are only some situations when the tabs can be different. I think that it's a good idea to move tabs in general, if are identical.

**Review by roxanariza — CHANGES_REQUESTED** [review]

File: src/contentMetaData.js (hunk 5 above)
Comment (adrian-musora) [reply src/contentMetaData.js]: I've moved common course information to common, good catch on the 'course' vs 'courses'

**Pull request merged**